src/CSModule.sol

```diff
@@ -1753,14 +1753,12 @@ contract CSModule is
         NodeOperator storage no = _nodeOperators[nodeOperatorId];
 
         uint256 newCount = no.totalVettedKeys - no.totalDepositedKeys;
-
         uint256 unbondedKeys = accounting.getUnbondedKeysCount(nodeOperatorId);
-        if (unbondedKeys > newCount) {
+
+        if (unbondedKeys >= no.totalAddedKeys - no.totalDepositedKeys) {
             newCount = 0;
-        } else {
-            unchecked {
-                newCount -= unbondedKeys;
-            }
+        } else if (unbondedKeys > no.totalAddedKeys - no.totalVettedKeys) {
+            newCount = no.totalAddedKeys - no.totalDepositedKeys - unbondedKeys;
         }
```

Comment:
I would suggest doing it like so to save extra bytes of bytecode:

```solidity
{
    uint256 availableKeys = no.totalAddedKeys - no.totalDepositedKeys;
    if (unbondedKeys >= availableKeys) {
        newCount = 0;
    } else if (unbondedKeys > no.totalAddedKeys - no.totalVettedKeys) {
        newCount = availableKeys - unbondedKeys;
    }
}
```

Reply:
-62 -> -33 bytecode size difference

Reply:
Done